File: task-03/JavaScript/Sub-task 3/sub-task-3.js
const readline = require("readline").createInterface({
    input: process.stdin,
    output: process.stdout,
});

readline.question("Enter the value of n: ", (input) => {
    const n = parseInt(input);
    pattern(n)
    readline.close();
});



function pattern(n){
	for (let i = 0; i < n; i++) {

        for (let j = i; j < n; j++) {
            process.stdout.write("  ");
        }

        for (let j = 0; j <= i; j++) {
            process.stdout.write("* ");
        }

        for (let j = 0; j < i; j++) {
            process.stdout.write("* ");
        }
        console.log();
    }

    for (let i = 0; i <= n; i++) {

       for (let j = 0; j < i; j++) {
            process.stdout.write("  ");
        }

        for (let j = i; j < n; j++) {
            process.stdout.write("* ");
        }

        for (let j = i; j <= n; j++) {
            process.stdout.write("* ");
        }
        console.log();
    }


}

File: task-03/Rust/Sub-task 3/sub-task-3.rs
use std::io;

fn main() {
    // Read the value of n from the user
    let mut input = String::new();
    println!("Enter the value of n: ");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Please enter a valid number");

    // First pattern
    for i in 0..n {
        for _ in i..n {
            print!("  ");
        }
        for _ in 0..=i {
            print!("* ");
        }
        for _ in 0..i {
            print!("* ");
        }
        println!();
    }

    // Second pattern
    for i in 0..=n {
        for _ in 0..i {
            print!("  ");
        }
        for _ in i..n {
            print!("* ");
        }
        for _ in i..=n {
            print!("* ");
        }
        println!();
    }
}
